docs/source/maintenance.rst

```diff
@@ -0,0 +1,164 @@
+Docs Maintenance
+================
+
+Even though the content of this documentation is generated automatically, its structure is handwritten and, unless we use custom sphinx templates in the future, must be maintained.
+
+This section shall serve as a short guide on how to maintaine this documentation.
+
+   * :ref:`The structure <structure>`
+   * :ref:`Adding or deleting a class or a (public) function to a module <simpleadd>`
+   * :ref:`Adding or deleting a submodule <heavyadd>`
+   * :ref:`Changing a class or a function within module <nochange>`
+   
+.. _structure:
+
+The structure
+-------------
+
+All of Psydac's documentation is contained in the /docs folder.
+This /docs folder consists of one directory, /source, as well as an empty .nojekyll file, a Makefile and make.bat.
+
+For the sake of maintenance, only the /source folder is of interest.
+
+docs/source
+###########
+
+This folder gives structure to the documentation. As this structure is self made, it needs to be maintained as Psydac grows.
+
+   * Every Sphinx documentation must contain an index.rst. Sphinx will generate index.html from this file, the index page of this documentation.
+   * This index.rst file must contain a toctree (.. toctree::) containing all "content" that is not included on the index page itself. As for now, 06.06.2023, this amounts to three sections, namely modules, examples and maintenance. Each of these sections has its own .rst file, modules.rst, examples.rst and maintenance.rst respectively. All .rst files which don't through toctrees have a path to this root toctree will not be part of the documentation.
```

Comment:
What do you mean with "don't through toctrees"? It seems to me that we can remove the words "through toctrees".

Reply:
I was trying to explain that even though for example docs/source/modules/linalg/solvers.rst is not included in the "root toctree" located in docs/source/index.rst, it is still included in the documentation as there is "a path to the root toctree", i.e., in docs/source/modules/linalg.rst there is a toctree containing .../solvers.rst. And the .../linalg.rst is contained in the docs/source/modules.rst toctree, and .../modules.rst is included in the "root toctree".

As you can tell, I am certainly lacking the right words to explain what I mean :')

Anyways, it's probably clear what is meant even without "through toctrees" and I will remove it.

Reply:
So you want to say that the "root toctree" does not have to point to every single module, but it could just point to subpackages of Psydac. For each subpackage one can then provide a "local toctree" which points to the modules to be documented in that subpackage. Correct?

Reply:
Yes exactly!

Reply:
3fa4e0272e920da64a042cbb87deb679ade437c5